bodhi/server/consumers/pungi_wrapper.py

```diff
@@ -0,0 +1,348 @@
+        for variant in self.compose.get_variants():
+            for arch in variant.arches + ["src"]:
+                timestamp = pungi.metadata.write_discinfo(self.compose, arch, variant)
+                pungi.metadata.write_media_repo(self.compose, arch, variant, timestamp)
+
+    def create_latest_repo_links(self):
+        latest_link = True
+
+        if latest_link:
+            self.compose_dir = os.path.basename(self.compose.topdir)
+            symlink_name = "latest-%s-%s" % (
+                self.compose.conf["release_short"],
+                self.compose.conf["release_version"]
+            )
+            if self.compose.conf["release_is_layered"]:
+                symlink_name += "-%s-%s" % (
+                    self.compose.conf["base_product_short"],
+                    self.compose.conf["base_product_version"]
+                )
+            symlink = os.path.join(self.compose.topdir, "..", symlink_name)
+
+            try:
+                os.unlink(symlink)
+            except OSError as ex:
+                if ex.errno != 2:
+                    raise Exception(ex)
+            try:
+                os.symlink(self.compose_dir, symlink)
+            except Exception as ex:
+                self.compose.log_error("Couldn't create latest symlink: %s" % ex)
+
+
+class VariantsConfig(object):
+
+    """
+    This class generates a variants config which can be used with the compose
+    object to generate a dnf repo. Right now it only works with modules.
+    """
+
+    def __init__(self, updates, builds, variant_id="Server",
+                 arches=['x86_64', 'armhfp', 'aarch64', 'i386', 'ppc64', 'ppc64le', 's390x']):
+        """ TODO: to be defined1. """
+        self.updates = updates
+        self.builds = builds
+        self.modules = self._generate_module_list()
+        self.arches = arches
+        self.headers = [
+            '<?xml version="1.0" encoding="UTF-8"?>',
+            ('<!DOCTYPE variants PUBLIC "-//Red Hat, Inc.//DTD '
+             'Variants info//EN" "variants2012.dtd">'),
+        ]
+        self.body = [
+            '<variants>',
+            '<variant id="%s" name="%s" type="variant">' % (variant_id, variant_id),
+            '<arches>',
+            ''.join(['<arch>%s</arch>' % arch for arch in self.arches]),
+            '</arches>',
+            '<modules>',
+            ''.join(['<module>%s</module>' % module for module in self.modules]),
+            '</modules>',
+            '</variant>',
+            '</variants>'
+        ]
+
+    def _generate_module_list(self):
+        newest_builds = {}
+        # we loop through builds so we get rid of older builds and get only
+        # a dict with the newest builds
+        for build in self.builds:
+            nsv = build.nvr.rsplit('-', 1)
+            ns = nsv[0]
+            version = nsv[1]
+
+            if ns in newest_builds:
+                curr_version = newest_builds[ns]
+                if int(curr_version) < int(version):
```

Comment:
Is the version guaranteed to be ```int```able? RPM version strings are not guaranteed to be integers.

Reply:
Keep in mind that versions often have "."'s in them, most often 2. ```int``` will blow up on those.

Reply:
for example: installer-f26-20170515150953
name: installer
stream: f26
version: 20170515150953

Version is a timestamp and always in this format.

Reply:
Yes, [it is an int](https://pagure.io/modulemd/blob/master/f/spec.yaml#_14).